src/cubicbez.rs

```diff
@@ -343,11 +537,34 @@ impl Iterator for ToQuads {
     }
 }
 
+/// Convert multiple cubic Bézier curves to quadratic splines.
+///
+/// Ensures that the resulting splines have the same number of control points.
+pub fn cubics_to_quadratic_splines(curves: &[CubicBez], accuracy: f64) -> Option<Vec<QuadSpline>> {
+    let mut splines: Vec<Option<QuadSpline>> = vec![None; curves.len()];
+    let mut last_unsuccessful_i = 0;
+    let mut split_order = 1;
+    let mut i = 0;
+    while split_order <= MAX_SPLINE_SPLIT {
+        if let Some(spline) = curves[i].approx_spline_n(split_order, accuracy) {
+            splines[i] = Some(spline);
+            i = (i + 1) % curves.len();
+            if i == last_unsuccessful_i {
+                return splines.into_iter().collect::<Option<Vec<_>>>();
```

Comment:
just to make sure I understand the logic:

- for each number of splits up to MAX, we try to generate a quadratic from the input
- if we succeed for each input, for a given split count, we return the quads
- if we fail for any input at a given split count we try the next split count
- if we try them all and fail, we bail?

Assuming that is true:
- I might prefer a `Result` to an `Option`, but not zealously
- I wonder if there's a more idiomatic approach? Maybe something like,

```rust
pub fn cubics_to_quadratic_splines(curves: &[CubicBez], accuracy: f64) -> Option<Vec<QuadSpline>> {
    let mut result = Vec::new();
    let mut split_order = 0;

    while split_order <= MAX_SPLINE_SPLIT {
        split_order += 1;
        result.clear();

        for curve in curves {
            match curve.approx_spline_n(split_order, accuracy) {
                Some(spline) => result.push(spline),
                None => break,
            }
        }

        if result.len() == curves.len() {
            return Some(result);
        }
    }
    None
}
```

partially this is about 'feel', but there are also potential performance costs to indexing into arrays in a tight loop, since each index op can generate a bounds check, and using iterators avoids this.

Reply:
You have the logic right. I’ll try reworking with your approach.